crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -504,28 +505,24 @@ pub(super) struct UseDefMapBuilder<'db> {
 
     /// Currently live bindings and declarations for each symbol.
     symbol_states: IndexVec<ScopedSymbolId, SymbolState>,
-
-    reachable: bool,
```

Comment:
At first I replaced this with `reachability: ScopedVisibilityConstraintId`, but I realized that `scope_start_visibility` is already what we need: the start of the scope is visible iff the flow is still reachable.